test/Makefile

```diff
@@ -151,9 +151,12 @@ build: download $(WASMS)
 
 $(WASMS): | $(OBJDIRS)
 $(OBJDIR)/%.wasm: $(OBJDIR)/%.wasm.o $(INFRA_WASM_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
+	$(CC) $(CFLAGS) $(LDFLAGS) -o tmp.wasm $^
```

Comment:
Its kind of a shame that we don't have a different file extension for components and core modules.. then we could just use two different rules here.  e.g. one rule for linking core modules from object files, and second rule that transforms wasm modules into components (based only file extension).

Is there a plan for a separate extension?    Should we just invent one here?   Perhaps we could use `foo.core.wasm` for the core module?

Reply:
I'm not aware of a plan for a separate extension.  Yes, we could invent one.  I guess splitting this rule into two implies that we'd also need two variants of the `%.wasm.err: %.wasm` rule below, one for when we want to run the component and the other for when we want to run the core module?

Reply:
That sounds cleaner to me yes.. especially since the command we use to invoke core modules vs components could looks very different.

Reply:
Was chatting with @sunfishcode and he suggested that using `.core.wasm` here some reasonable.   (Or we could invert it and give components a special name such as `.cmp.wasm` or `.wasm.cmp`).

Reply:
OK, I just pushed an update.  In order to test it properly, I had to update the Wasmtime URL from the (constantly updating) `dev` release to the `v17.0.0` release, which in turn required updating the bindings to the final WASI `0.2.0` release, so the PR has ballooned.  Apologies for that.

Reply:
Do you want to update the PR title and description to include all this new stuff?  Or make a separate PR?

Reply:
I've updated the title and description.